blocks/library/image/style.scss

```diff
@@ -9,11 +9,19 @@
 	&[data-align="left"] {
 		float: left;
 		margin-right: $block-padding;
+
+		@include break-medium() {
+			margin-left: calc( 50% - #{ $visual-editor-max-width / 2 } );
+		}
```

Comment:
Aside: Do we expect block authors to implement this CSS styles (which are not that easy) in each floated block